src/hid/joystick_glfw_keyboard.cc

```diff
@@ -0,0 +1,60 @@
+#include "hid/joystick_glfw_keyboard.h"
+
+// GLFW
+#include <GLFW/glfw3.h>
+
+namespace BoBRobotics {
+namespace HID {
+
+//! Open default joystick device with (optionally) specified dead zone
+JoystickGLFWKeyboard::JoystickGLFWKeyboard(GLFWwindow *window)
+  : JoystickBase(0.0f), m_Window(window)
+{
+
+}
+//------------------------------------------------------------------------
+bool JoystickGLFWKeyboard::updateState()
+{
+    // Use WSAD for left stick axes
+    setAxisStateKeys(GLFW_KEY_W, GLFW_KEY_S, JAxis::LeftStickVertical);
+    setAxisStateKeys(GLFW_KEY_A, GLFW_KEY_D, JAxis::LeftStickHorizontal);
+
+    // Use cursor for right stick axes
+    setAxisStateKeys(GLFW_KEY_UP, GLFW_KEY_DOWN, JAxis::RightStickVertical);
+    setAxisStateKeys(GLFW_KEY_LEFT, GLFW_KEY_RIGHT, JAxis::RightStickHorizontal);
+
+    // Use page up and down for right trigger
+    setAxisStateKeys(GLFW_KEY_PAGE_UP, GLFW_KEY_PAGE_DOWN, JAxis::RightTrigger);
+
+    setButtonStateKey(GLFW_KEY_1, JButton::A);
+    setButtonStateKey(GLFW_KEY_2, JButton::B);
+    setButtonStateKey(GLFW_KEY_3, JButton::X);
+    setButtonStateKey(GLFW_KEY_4, JButton::Y);
+
+    return true;
```

Comment:
Maybe in a future PR we could make this configurable, e.g. with a YAML file, because we're using some of the other buttons elsewhere and it might also be able to configure what axis value is used when a keyboard button is pressed

Reply:
Good idea - opened #143